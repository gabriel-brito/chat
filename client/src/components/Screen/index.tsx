import Message from 'components/Message'
import * as S from './styles'

type mockMessageProps = {
  message: string
  user: string
  id: string
}

export type ScreenProps = {
  messages?: mockMessageProps[]
  userId?: string
}

const Screen = ({ messages, userId }: ScreenProps) => (
  <S.Wrapper>
    {messages?.map(({ message, id, user }, index) => {
      const isFromUser = id === userId

      return (
        <S.MessageWrapper fromUser={isFromUser} key={index}>
          <Message fromUser={isFromUser}>
            {!isFromUser && `${user}: `}
            {message}
          </Message>
        </S.MessageWrapper>
      )
    })}
  </S.Wrapper>
)

export default Screen
