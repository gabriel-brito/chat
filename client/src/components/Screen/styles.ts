import { MessageProps } from 'components/Message'
import styled, { css } from 'styled-components'

export const Wrapper = styled.ul`
  background-image: url('/img/chat-wallpaper.png');
  background-size: cover;
  display: flex;
  flex-direction: column;
  height: 100%;
  overflow: auto;
  padding: 1em;
  width: 100%;
`

export const MessageWrapper = styled.li<Pick<MessageProps, 'fromUser'>>`
  ${({ fromUser }) => css`
    display: flex;
    justify-content: ${fromUser ? 'flex-end' : 'flex-start'};
    list-style: none;
    margin-bottom: 1em;
    width: 100%;
  `}
`
