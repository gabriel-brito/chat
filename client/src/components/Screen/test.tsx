import { render } from '@testing-library/react'

import Screen from '.'

const messages = [
  { message: 'Hello there!', id: 'id-1', user: 'Gabriel' },
  { message: 'Hello!', id: 'id-2', user: 'Bot-1567' },
  { message: 'How are you doing today?', id: 'id-1', user: 'Gabriel' },
  {
    message: `I'm really good! and what about you?`,
    id: 'id-2',
    user: 'Bot-1567'
  },
  { message: `I'm great, thanks!`, id: 'id-1', user: 'Gabriel' },
  {
    message: `There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable. If you are going to use a passage of Lorem Ipsum, you need to be sure there isn't anything embarrassing hidden in the middle of text. All the Lorem Ipsum generators on the Internet tend to repeat predefined chunks as necessary, making this the first true generator on the Internet. It uses a dictionary of over 200 Latin words, combined with a handful of model sentence structures, to generate Lorem Ipsum which looks reasonable. The generated Lorem Ipsum is therefore always free from repetition, injected humour, or non-characteristic words etc.`,
    id: 'id-2',
    user: 'Bot-1567'
  }
]

describe('<Screen />', () => {
  it('should render the Screen and match Snapshot', () => {
    const { container } = render(<Screen />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render a list if messages is not null', () => {
    const { getAllByRole } = render(<Screen messages={messages} />)

    const listItems = getAllByRole('listitem')

    expect(listItems).toHaveLength(6)
  })

  it('should render flex-end if the message is from the user', () => {
    const { getAllByRole } = render(
      <Screen userId="id-1" messages={messages} />
    )

    const firstMessage = getAllByRole('listitem')[0]

    expect(firstMessage).toHaveStyle({ justifyContent: 'flex-end' })
  })
})
