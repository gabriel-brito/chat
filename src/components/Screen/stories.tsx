import { Story, Meta } from '@storybook/react/types-6-0'
import Screen, { ScreenProps } from '.'
import { v4 } from 'uuid'

export default {
  title: 'Screen',
  component: Screen
} as Meta

const messages = [
  { message: 'Hello there!', user: 'Gabriel' },
  { message: 'Hello!', id: 'id-2', user: 'Bot-1567' },
  { message: 'How are you doing today?', user: 'Gabriel' },
  {
    message: `I'm really good! and what about you?`,
    id: 'id-2',
    user: 'Bot-1567'
  },
  { message: `I'm great, thanks!`, user: 'Gabriel' },
  {
    message: `There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable. If you are going to use a passage of Lorem Ipsum, you need to be sure there isn't anything embarrassing hidden in the middle of text. All the Lorem Ipsum generators on the Internet tend to repeat predefined chunks as necessary, making this the first true generator on the Internet. It uses a dictionary of over 200 Latin words, combined with a handful of model sentence structures, to generate Lorem Ipsum which looks reasonable. The generated Lorem Ipsum is therefore always free from repetition, injected humour, or non-characteristic words etc.`,
    id: 'id-2',
    user: 'Bot-1567'
  }
]

export const Default: Story<ScreenProps> = () => (
  <div style={{ maxWidth: '35em', margin: '0 auto', height: '100vh' }}>
    <Screen />
  </div>
)

export const WithMessages: Story<ScreenProps> = () => (
  <div style={{ maxWidth: '35em', margin: '0 auto', height: '100vh' }}>
    <Screen nickname={'Gabriel'} messages={messages} />
  </div>
)
