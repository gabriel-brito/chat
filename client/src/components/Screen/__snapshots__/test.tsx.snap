// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Screen /> should render the Screen and match Snapshot 1`] = `
.c0 {
  background-image: url('/img/chat-wallpaper.png');
  background-size: cover;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  overflow: scroll;
  padding: 1em;
  width: 100%;
}

<ul
  class="c0"
/>
`;
